From 0bc307714c7fe8a37e5cf9632fbe4eb88e439c83 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Thu, 4 Oct 2018 11:31:28 +0300
Subject: [PATCH 4833/5242] MLK-18497-2: ASoC: fsl: dsp_proxy: Add missing
 xf_unlock

commit  28174671738a65ca3645a69a0038bc1865ffa434 from
https://source.codeaurora.org/external/imx/linux-imx.git

xf_cmd_send_recv function returns with a lock taken
in case of success. Fix this, now!

This bug is present since the beginning of time and it didn't
show up because no one used xd_cmd_alloc/xf_cmd_free.

Reviewed-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp_proxy.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/fsl/fsl_dsp_proxy.c b/sound/soc/fsl/fsl_dsp_proxy.c
index 30e5f81..bc07158 100644
--- a/sound/soc/fsl/fsl_dsp_proxy.c
+++ b/sound/soc/fsl/fsl_dsp_proxy.c
@@ -615,6 +615,7 @@ int xf_cmd_alloc(struct xf_proxy *proxy, void **buffer, u32 length)
 
 	/* ...free message and release proxy lock */
 	xf_msg_free(proxy, m);
+	xf_unlock(&proxy->lock);
 
 	return ret;
 }
@@ -641,6 +642,7 @@ int xf_cmd_free(struct xf_proxy *proxy, void *buffer, u32 length)
 
 	/* ...free message and release proxy lock */
 	xf_msg_free(proxy, m);
+	xf_unlock(&proxy->lock);
 
 	return ret;
 }
-- 
1.7.9.5

